core/vm/intpool.go

```diff
@@ -65,3 +68,45 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolDefaultCap = 5
+
+// IntPoolPool manages a pool of intPools
+type intPoolPool struct {
+	pools []*intPool
+
+	// Mutex
+	mutex sync.Mutex
+}
+
+func initPool() {
+	poolOfIntPools = intPoolPool{
+		pools: make([]*intPool, 0, poolDefaultCap),
+	}
+}
+
+var poolOfIntPools intPoolPool
+
+// get is looking for an available pool to return
+func (ipp *intPoolPool) get() *intPool {
+	ipp.mutex.Lock()
+	defer ipp.mutex.Unlock()
+
+	if len(poolOfIntPools.pools) > 0 {
+		ip := ipp.pools[len(ipp.pools)-1]
+		ipp.pools = ipp.pools[:len(ipp.pools)-1]
+		return ip
+	}
+
+	return newIntPool()
+}
+
+// release a pool that has been allocated with Get
+func (ipp *intPoolPool) release(ip *intPool) {
+	ipp.mutex.Lock()
+	defer ipp.mutex.Unlock()
+
+	if len(ipp.pools) < poolDefaultCap {
```

Comment:
you could use `if len(ipp.pools) < cap(ipp.pools)` here and that would make the code more robust (i.e. if someone changes the constructor, this will keep in sync with it).